component/src/main/java/me/androidbox/component/general/PasswordEntry.kt

```diff
@@ -0,0 +1,112 @@
+package me.androidbox.component.general
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.*
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PasswordEntry(
+    modifier: Modifier = Modifier,
+    passwordValue: String,
+    placeholderText: String,
+    isPasswordVisible: MutableState<Boolean> = mutableStateOf(false),
```

Comment:
States should be passed as a field directly to your composables and not as mutable state. This would very likely cause lots of different sources from where the state could be mutated which makes it super hard to find bugs. The only place where the state should be changed is in the corresponding ViewModel.

Therefore, here you should just pass a normal boolean whether the password is visible or not. The logic for toggling it belongs in the ViewModel.

Reply:
@philipplackner 
Thanks, I am new to compose so your insight here is very valuable.